packages/component-base/src/data-provider-controller/data-provider-controller.d.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license
+ * Copyright (c) 2021 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import type { ReactiveController } from 'lit';
+import type { Cache } from './cache.js';
+
+type DataProviderParams = {
+  page: number;
+  pageSize: number;
+  parentItem?: unknown;
+  [key: string]: unknown;
+};
+
+type DataProviderCallback = (items: unknown[], size?: number) => void;
+
+type DataProvider = (params: DataProviderParams, callback: DataProviderCallback) => void;
+
+/**
+ * A controller that stores and manages items loaded with a data provider.
+ */
+export class DataProviderController implements ReactiveController {
```

Comment:
Getting this error

<img width="476" alt="Screenshot 2023-09-25 at 11 04 34" src="https://github.com/vaadin/web-components/assets/1222264/7f4bc3c0-da09-4236-bc44-597055989527">

Reply:
Fixed.